### DISCUSSION TITLE
split = "enew" does not really switch buffers right

### DISCUSSION THREAD

**Discussion by srcrip** [ibhagwan/fzf-lua#1440] [Q&A] [answered]
Hey all,

if you set winopts.split = "enew", make a new split, and then launch fzf-lua, I expect it to switch just that window to the new buffer, but it does not work. instead by split gets deleted, and I end up with only one window with the new buffer. Any idea why?

**Comment by ibhagwan** [comment]
I’m not sure but what exactly are you trying to achieve that isn’t possible with the documented split commands `botright split` etc?

**Reply by ibhagwan** [reply]
Is it possible you meant you’d like to open a buffer in a split? For that press ctrl-s (or ctrl-v for vsplit) instead of enter.

**Reply by srcrip** [reply]
what I want to do is:

1. first open a split, or put my cursor in an already splitted window
2. run :fzflua files or similar command
3. I set my `split` command to `enew`, so fzf opens a new buffer, which takes up the entirety of the split window
4. select a file
5. it should then open that file *in my split window* that fzf was just in

What happens right now, if you try setting `split` to `enew`, is everything works up to step 5, but when fzflua opens the file, it deletes the split. I feel like what's happening is that the fzflua buffer is not just killing the buffer but also trying to close the window. then the file opens in the remaining window. Does that make sense?

**Reply by ibhagwan** [reply]
> it should then open that file in my split window that fzf was just in

Not possible without custom code, however fzf-lua opens (split or float) it will always discard the terminal buffer it was open inside, you can perhaps simulate the behavior by opening in a split and then using the same split command to open the selected buffer (similar to ctrl-s) but it won’t be exactly the same as what you’re after.

**Comment by ibhagwan** [comment, marked as answer]
You can perhaps try `:FzfLua files autoclose=false` but you’ll have to handle closing the window after aborting with ctrl-c/esc.

**Reply by srcrip** [reply]
That actually works perfectly. I don't see that documented anywhere, is that new?

**Reply by srcrip** [reply]
also thank you very much for the prompt response. you're the best. and thanks for making fzflua.

**Reply by ibhagwan** [reply]
> That actually works perfectly. I don't see that documented anywhere, is that new?

I’m still working on documenting all esoteric options it’s “WIP” lol 

Note that when you exit without selecting you’ll have a “process closed” window that won’t close.

There are a few ways to go around that:
- Create esc/ctrl-c  custom actions that call nvim_win_close
- Do not use autoclose, instead change the mapping of enter to `{ fn =actions.file_edit, noclose = true }`.

**Reply by ibhagwan** [reply]
> also thank you very much for the prompt response. you're the best. and thanks for making fzflua.

Ty @srcrip <3
